src/pykiso/lib/auxiliaries/zephyr.py

```diff
@@ -0,0 +1,217 @@
+# Copyright (c) 2010-2022 Robert Bosch GmbH
+# This program and the accompanying materials are made available under the
+# terms of the Eclipse Public License 2.0 which is available at
+# http://www.eclipse.org/legal/epl-2.0.
+#
+# SPDX-License-Identifier: EPL-2.0
+##########################################################################
+import enum
+import logging
+import subprocess
+import xml.etree.ElementTree as ET
+from pathlib import Path
+from typing import Optional
+from pykiso import SimpleAuxiliaryInterface
+from pykiso.types import MsgType
+
+log = logging.getLogger(__name__)
+
+
+class TestResult(enum.IntEnum):
+    """Result of a twister test"""
+
+    PASSED = 0
+    FAILED = 1
+    SKIPPED = 2
+    ERROR = 3
+
+
+class ZephyrError(Exception):
+    """Exception for twister errors"""
+
+    pass
+
+
+class Twister:
+    """Control class for the Zephyr twister commandline tool"""
+
+    def __init__(self, twister_path: str = "twister") -> None:
+        self.twister_path = twister_path
+        self.process: Optional[subprocess.Popen[bytes]] = None
+
+    def start_test(
+        self, test_directory: str, testcase_name: str, wait_for_start: bool = True
+    ) -> None:
+        """Start the test case
+
+        :param test_directory: The directory to search for the Zephyr test project (passed to twister using the -T parameter)
+        :param testcase_name: The name of the Zephyr test (passed to twister using the -s parameter)
+        :param wait_for_start: Wait for the test case to start on the target. If this is set to false the function will just return even if the test is not running yet.
+
+        :return: True if successful
+        """
+        if self.process is not None:
+            raise ZephyrError("Twister test is already running")
+
+        self.outdir = str(Path(test_directory).resolve() / "twister-out")
+        logging.info(f"Using Twister output directory: {self.outdir}")
+
+        self.process = subprocess.Popen(
+            [
+                self.twister_path,
+                "-vv",
+                "-T",
+                test_directory,
+                "-s",
+                testcase_name,
+                "--outdir",
+                self.outdir,
+                "--report-dir",
+                self.outdir,
+            ],
+            stderr=subprocess.PIPE,
+            shell=False,
+        )
+
+        if wait_for_start:
+            # Read twister output until test case was started
+            while True:
+                line = self._readline()
+                if line == None:
+                    return
+
+                if "OUTPUT: START - " in line:
+                    log.info(f"Zephyr test started.")
+                    return
+
+    def _readline(self) -> Optional[str]:
+        """Read a line from the twister process
+
+        :return: The line as string or None if the process has finished
+        """
+        while True:
```

Comment:
sooo.... Can it get stuck? No timeout?

Reply:
If the twister tool gets stuck, this will also get stuck. Timeout would depend on the duration of the Zephyr test so we have to at least make it configurable.